reactivesocket-transport-tcp/src/main/java/io/reactivesocket/transport/tcp/client/TcpReactiveSocketConnector.java

```diff
@@ -1,81 +1,50 @@
-/**
- * Copyright 2016 Netflix, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package io.reactivesocket.transport.tcp.client;
 
-import io.netty.channel.EventLoopGroup;
-import io.reactivesocket.*;
-import io.reactivesocket.internal.rx.EmptySubscription;
-import io.reactivesocket.rx.Completable;
+import io.netty.buffer.ByteBuf;
+import io.reactivesocket.ConnectionSetupPayload;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.ReactiveSocketConnector;
+import io.reactivex.netty.protocol.tcp.client.TcpClient;
 import org.reactivestreams.Publisher;
-import org.reactivestreams.Subscriber;
-import org.reactivestreams.Subscription;
 
 import java.net.SocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Consumer;
+import java.util.function.Function;
 
-/**
- * An implementation of {@link ReactiveSocketConnector} that creates Netty TCP ReactiveSockets.
- */
 public class TcpReactiveSocketConnector implements ReactiveSocketConnector<SocketAddress> {
-    private final ConnectionSetupPayload connectionSetupPayload;
+
+    private final ConcurrentMap<SocketAddress, TcpReactiveSocketFactory> socketFactories;
```

Comment:
You are correct, for this change it is not required.

In the (slightly?)longer term, I think we don't need the connector as up the chain `LoadBalancer` keeps a list of `ReactiveSocketFactory` and never really calls `connect(SocketAddress)` which is the correct thing to do. The only reason `ReactiveSocketConnector` is required today because it provides the source (list of Sockets) in `ClientBuilder` to `ReactiveSocketFactory` mapping. 
I suggest that we use the source as `Publisher<List<ReactiveSocketFactory>>` which will remove the need of `ReactiveSocketConnector` and hence the ugly mapping inside _any_ `ReactiveSocketConnector` implementation as generally clients bind to an address. 
As I said, we can discuss this in a separate PR. 

If you feel strongly about removing this abstraction in this PR, I will remove it but I think `ReactiveSocketFactory` should be the correct abstraction which is bound to a single `SocketAddress`